platform/darwin/src/MGLCircleStyleLayer.mm

```diff
@@ -47,6 +47,14 @@ - (instancetype)initWithIdentifier:(NSString *)identifier source:(MGLSource *)so
     return self;
 }
 
+- (instancetype)initWithRawLayer:(mbgl::style::CircleLayer *)rawLayer source:(MGLSource *)source
+{
+    if (self = [super initWithIdentifier:@(rawLayer->getID().c_str()) source:source]) {
```

Comment:
The MGLSource can always be obtained from the `mbgl::style::Layer`, so we could move `-[MGLStyle sourceFromMBGLSource:]` to `+[MGLSource sourceWithRawSource:]` in MGLSource_Private.h (alongside the existing `-[MGLSource initWithRawSource:]`). That would make `-[MGLStyle layerFromMBGLLayer:]` a bit cleaner, since we wouldn’t need to pass in an MGLSource every time we wrap `mbgl::style::Layer`.

Not a big deal, but possibly worth doing the next time we have to touch this code.